test/integration/02-user-wallet/02-send-onchain.spec.ts

```diff
@@ -116,75 +124,113 @@ afterAll(async () => {
 
-    let pendingTxHash: OnChainTxHash
+  let pendingTxHash: OnChainTxHash
 
-    {
-      const txResult = await Wallets.getTransactionsForWalletId({
-        walletId: walletIdA,
-      })
-      if (txResult.error instanceof Error || txResult.result === null) {
-        throw txResult.error
-      }
-      const pendingTxs = txResult.result.filter(
-        ({ status }) => status === TxStatus.Pending,
-      )
-      expect(pendingTxs.length).toBe(1)
-      const pendingTx = pendingTxs[0]
-      expect(pendingTx.settlementAmount).toBe(-amount - pendingTx.settlementFee)
-      pendingTxHash = pendingTx.id as OnChainTxHash
+  {
+    const txResult = await Wallets.getTransactionsForWalletId({
+      walletId: senderWalletId,
+    })
+    if (txResult.error instanceof Error || txResult.result === null) {
+      return txResult.error
+    }
+    const pendingTxs = txResult.result.filter(({ status }) => status === TxStatus.Pending)
+    expect(pendingTxs.length).toBe(1)
+    const pendingTx = pendingTxs[0]
+    const interimBalance = await getBalanceHelper(senderWalletId)
 
-      const interimBalance = await getBalanceHelper(walletIdA)
-      expect(interimBalance).toBe(initialBalanceUserA - amount - pendingTx.settlementFee)
-      await checkIsBalanced()
+    if (sendAll) {
+      expect(pendingTx.settlementAmount).toBe(-initialWalletBalance)
+      expect(interimBalance).toBe(0)
+    } else {
+      expect(pendingTx.settlementAmount).toBe(-amount - pendingTx.settlementFee)
+      expect(interimBalance).toBe(initialWalletBalance - amount - pendingTx.settlementFee)
     }
 
-    // const subSpend = subscribeToChainSpend({ lnd: lndonchain, bech32_address: address, min_height: 1 })
+    pendingTxHash = pendingTx.id as OnChainTxHash
+
+    await checkIsBalanced()
+  }
+
+  // const subSpend = subscribeToChainSpend({ lnd: lndonchain, bech32_address: address, min_height: 1 })
 
-    await Promise.all([once(sub, "chain_transaction"), mineBlockAndSyncAll()])
+  await Promise.all([
+    once(sub, "chain_transaction"),
+    mineBlockAndSync({ lnds: [lndonchain] }),
+  ])
 
-    await sleep(3000)
+  await sleep(1000)
 
+  expect(sendNotification.mock.calls.length).toBe(1)
+
+  const senderWallet = await WalletsRepository().findById(senderWalletId)
+  if (senderWallet instanceof Error) return senderWallet
+
+  if (!sendAll) {
     const satsPrice = await Prices.getCurrentPrice()
-    if (satsPrice instanceof Error) throw satsPrice
+    if (satsPrice instanceof Error) return satsPrice
 
-    const paymentAmount = { amount: BigInt(amount), currency: WalletCurrency.Btc }
+    const paymentAmount = { amount: BigInt(amount), currency: senderWallet.currency }
     const displayPaymentAmount = {
-      amount: amount * satsPrice,
+      amount: senderWallet.currency === WalletCurrency.Btc ? amount * satsPrice : amount,
       currency: DefaultDisplayCurrency,
     }
 
```

Comment:
`createPushNotificationContent` is only in the non send all branch. there is no notification for sendAll?

Reply:
Good catch, I was supposed to circle back to that. Fixed https://github.com/GaloyMoney/galoy/pull/2072/commits/e3848a1d8c84f8483ce46c823c8b279674092196